rundeckapp/src/test/groovy/rundeck/services/ProjectManagerServiceSpec.groovy

```diff
@@ -721,6 +725,38 @@ class ProjectManagerServiceSpec extends Specification implements ServiceUnitTest
         "789"==result.getProperty("ghi")
     }
 
+    void "merge project properties"(){
+        given:
+        RundeckProject rundeckProject = new RundeckProject("project", null, service)
+        RundeckProjectDataProvider providerMock = Mock(RundeckProjectDataProvider){
+            findByName("project") >> Mock(RdProject)
+        }
+        service.configStorageService=Stub(ConfigStorageService){
+            existsFileResource("projects/test1/my-resource") >> true
+            existsFileResource("projects/test1/not-my-resource") >> false
+        }
+
+        service.projectDataProvider = providerMock
+
+        def properties = new Properties()
+        properties.setProperty("fwkprop","fwkvalue")
+        properties.setProperty("project.description", "desc")
+
+        service.frameworkService=Stub(FrameworkService){
+            getRundeckFramework() >> Stub(Framework){
+                getPropertyLookup() >> PropertyLookup.create(properties)
+            }
+        }
+        service.rundeckNodeService=Mock(NodeService)
+
+        when:
+        service.mergeProjectProperties(rundeckProject, properties, null)
+
+        then:
+        rundeckProject.projectConfig.name == "project"
+        rundeckProject.nodesFactory
```

Comment:
do we expect that `rundeckProject.nodesFactory` will be set? maybe we can that more explicit (like `rundeckProject.nodesFactory!=null`)

Reply:
Actually it expects to be equals to `rundeckNodeService`. Just updated it